SQF-VM/SQF_commands.c

```diff
@@ -45,6 +45,35 @@ int64_t system_time_ms(void)
 	return (int64_t)(((timebuffer.time * 1000) + timebuffer.millitm));
 #endif
 }
+
+/**
+ * Checks whether the given array is a 3D vector.
+ * If not an error will be thrown
+ */
+char checkVector3(const PVM vm, const PARRAY array)
```

Comment:
Advice for better reusability:
- Introduce new parameter `index` setting the start index of the range to check
- Introduce new parameter `length` setting the range to check (defaults to 3 right now)
- Introduce new parameter `type` providing the type that is expected to be in range
- Rename method to `check_array_range`

using macros, it then is possible to branch "commands" like `check_array_is_vector3`